packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -142,12 +174,42 @@ class AnalyzeCommand extends FlutterCommand {
       return false;
     }
 
+    // Don't run pub if asking for android analysis.
+    if (boolArg('android')) {
+      return false;
+    }
+
     return super.shouldRunPub;
   }
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    if (boolArg('suggestions')) {
+    if (boolArg('android')) {
+      final AndroidAnalyzeOption option;
+      if (argResults!['list-build-variants'] as bool) {
+        option = AndroidAnalyzeOption.listBuildVariant;
+      } else if (argResults!['output-app-link-settings'] as bool) {
```

Comment:
should we also throw if both of these are passed? Since it seems like we will silently choose `list-build-variants` and ignore `output-app-link-settings` here.